src/hotspot/os/posix/signals_posix.hpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef OS_POSIX_SIGNALS_POSIX_HPP
+#define OS_POSIX_SIGNALS_POSIX_HPP
+
+#include "memory/allocation.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+#include <signal.h>
+
+// Signal number used to suspend/resume a thread
+// do not use any signal number less than SIGSEGV, see 4355769
+static int SR_signum = SIGUSR2;
```

Comment:
Can you hide this in the .cpp file so that you can avoid including <signal.h> in the header file?

And use forward declarations for outputStream, ucontext_t, and siginfo_t.

Reply:
Note that these changes are not important enough to rerun testing for this cleanup, so you can make it a new RFE.  Thank you for doing consolidation!

Reply:
SR_signum is referenced from the os_*.cpp files so needs to be in the header file now.

Reply:
I wondered why this is not const then I saw that it can be overwritten with an environment variable "_JAVA_SR_SIGNUM". Is this still needed? Seems to be an odd way of specifying a VM parameter. And it seems to be untested (I did not find any tests for that in the open codebase).